src/utils.ts

```diff
@@ -232,6 +232,87 @@ const mergeObjects = <T extends object>(...objects: T[]): T => {
 const moveElementToEnd = <T>(array: T[], index: number): T[] =>
     array.slice(0, index).concat(array.slice(index + 1), array[index]);
 
+const queryStringParser = (url: string, keys: string[]): Dictionary<string> => {
+    let urlParams;
+    let results = {};
+
+    if (!url) return results;
+
+    if (typeof URL !== 'undefined' && typeof URLSearchParams !== 'undefined') {
+        const urlObject = new URL(url);
+        urlParams = new URLSearchParams(urlObject.search);
+    } else {
+        urlParams = queryStringParserFallback(url);
+    }
+
+    keys.forEach(key => {
+        const value = urlParams.get(key);
+        if (value) {
+            results[key] = value;
+        }
+    });
+
+    return results;
+};
+
+interface URLSearchParamsFallback {
+    get: (key: string) => string | null;
+}
+
+const queryStringParserFallback = (url: string): URLSearchParamsFallback => {
+    var params = {};
+    var queryString = url.split('?')[1];
+    var pairs = queryString.split('&');
+
+    pairs.forEach(pair => {
+        var [key, value] = pair.split('=');
+        if (key && value) {
+            params[key] = decodeURIComponent(value || '');
+        }
+    });
+
+    return {
+        get: function(key: string) {
+            return params[key];
+        },
+    };
+};
+
+// Get cookies as a dictionary
+const getCookies = (keys?: string[]): Dictionary<string> => {
```

Comment:
`getCookies` seems to generically named given it is in a `utils` file.  Just given the name, it I would have expected it to just return cookies.  However, it is doing a lot more than just getting the cookies.  I think this is okay for now, but it's not clear to me if this method can be used elsewhere where we are getting and parsing cookies.

This is also true of queryString methods here.  They seem pretty specific to the integrationId feature, rather than being generic methods that are reusable.

Thoughts?